obspy/io/gse2/bulletin.py

```diff
@@ -0,0 +1,1117 @@
+                'ev_az': slice(13, 18),
+                'picktype': slice(19, 20),
+                'direction': slice(20, 21),
+                'detchar': slice(21, 22),
+                'phase': slice(23, 30),
+                'time': slice(31, 52),
+                't_res': slice(53, 58),
+                'azim': slice(59, 64),
+                'az_res': slice(65, 71),
+                'slow': slice(72, 77),
+                's_res': slice(78, 83),
+                't_def': slice(84, 85),
+                'a_def': slice(85, 86),
+                's_def': slice(86, 87),
+                'snr': slice(88, 93),
+                'amp': slice(94, 103),
+                'per': slice(104, 109),
+                'mag_type_1': slice(110, 112),
+                'mag_1': slice(112, 116),
+                'mag_type_2': slice(117, 119),
+                'mag_2': slice(119, 123),
+                'id': slice(124, 132),
+            },
+        }
+
+        if fields:
+            if 'line_1' in fields:
+                self.fields['line_1'].update(fields['line_1'])
+            if 'line_2' in fields:
+                self.fields['line_2'].update(fields['line_2'])
+            if 'arrival' in fields:
+                self.fields['arrival'].update(fields['arrival'])
+
+    def load(self, filename):
+        """
+        Read GSE2.0 file into ObsPy catalog object.
+
+        :type filename: str
+        :param filename: File name to read.
+        :rtype: :class:`~obspy.core.event.Catalog`
+        :return: ObsPy Catalog object.
+        """
+        with open(filename, 'r') as f:
+            self.lines = LinesIterator(f.readlines())
+        return self._deserialize()
+
+    def _add_line_nb(self, message):
+        """
+        Add line number at the end of a str message.
+
+        :type message: str
+        :param message: Message for warnings or exceptions.
+        :rtype: str
+        :return: Message with line number.
+        """
+        return "%s, line %s" % (message, self.lines.line_nb)
+
+    def _warn(self, message):
+        warnings.warn(self._add_line_nb(message))
+
+    def _skip_empty_lines(self):
+        line = next(self.lines)
+        while not line or line.isspace():
+            line = next(self.lines)
+        return line
+
+    def _get_res_id(self, ident, parent=None, parent_res_id=None):
+        """
+        Create a :class:`~obspy.core.event.base.ResourceIdentifier` object.
+
+        :type ident: str
+        :param ident: Id of
+            the :class:`~obspy.core.event.base.ResourceIdentifier`.
+        :type parent: :class:`~obspy.core.event.origin.Origin`,
+            :class:`~obspy.core.event.Event` or any other object with a
+            resource_id attribute.
+        :param parent: The resource_id attribute of the parent will be
+            used as a prefix for the new
+            :class:`~obspy.core.event.base.ResourceIdentifier`.
+        :type parent_res_id: :class:`~obspy.core.event.ResourceIdentifier`
```

Comment:
Has to be `~obspy.core.event.base.ResourceIdentifier` (as is done above). The same is true for the other references to this further down this docstring.